selfdrive/ui/qt/window.cc

```diff
@@ -4,24 +4,26 @@
 
 #include "system/hardware/hw.h"
 
-MainWindow::MainWindow(QWidget *parent) : QWidget(parent) {
+MainWindow::MainWindow(QWidget *parent, HomeWindow *hw, SettingsWindow *sw, OnboardingWindow *ow) :
+    QWidget(parent),
+    homeWindow(hw ? hw : new HomeWindow(this)),
+    settingsWindow(sw ? sw : new SettingsWindow(this)),
+    onboardingWindow(ow ? ow : new OnboardingWindow(this)) {
```

Comment:
I dont recall when we had to pass this windows like this so it's a bit confusing. I tried looking at our old code and I was just as confused. We may need a comment or refresher. Im probably the one to blame here lmao

Reply:
Reading at the header `.h` it kinda clarifies a bit. 
```suggestion
// We have this constructor so that we can provide custom implementations of the windows. By default (stock_ui) would receive them as nullptr, so they'll be instantiated with stock. Otherwise they'd be SP instances
MainWindow::MainWindow(QWidget *parent, HomeWindow *hw, SettingsWindow *sw, OnboardingWindow *ow) :
    QWidget(parent),
    homeWindow(hw ? hw : new HomeWindow(this)),
    settingsWindow(sw ? sw : new SettingsWindow(this)),
    onboardingWindow(ow ? ow : new OnboardingWindow(this)) {
```